consensus/enclave/api/src/lib.rs

```diff
@@ -93,14 +93,26 @@ impl WellFormedTxContext {
         }
     }
 
+    /// Create a new WellFormedTxContext, from a Tx and its priority.
+    pub fn new_from_tx(priority: u64, tx: &Tx) -> Self {
```

Comment:
Optional/Equivalnet: Omit `new_`, maybe swap the args

```suggestion
    pub fn from_tx(tx: &Tx, priority: u64) -> Self {
```